partner_email_duplicate_warn/models/res_partner.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    same_email_partner_id = fields.Many2one(
+        "res.partner",
+        compute="_compute_same_email_partner_id",
+        string="Partner with same e-mail",
+        compute_sudo=True,
+    )
+
+    @api.depends("email", "company_id")
+    def _compute_same_email_partner_id(self):
+        for partner in self:
+            same_email_partner_id = False
+            if partner.email and partner.email.strip():
+                partner_email = partner.email.strip().lower()
+                domain = [("email", "=ilike", "%" + partner_email + "%")]
+                if partner.company_id:
+                    domain += [
+                        "|",
+                        ("company_id", "=", False),
+                        ("company_id", "=", partner.company_id.id),
+                    ]
+                partner_id = partner._origin.id
+                if partner_id:
+                    domain += [
+                        ("id", "!=", partner_id),
+                        "!",
+                        ("id", "child_of", partner_id),
+                        "!",
+                        ("id", "parent_of", partner_id),
+                    ]
+                search_partners = self.with_context(active_test=False).search(domain)
```

Comment:
Maybe I am wrong but in the search_partners result all email will match and will have a value ?
if yes why not just using 

```
self.with_context(active_test=False).search(domain, limit=1)
```

Reply:
Thus 
`partner.same_email_partner_id = self.with_context(active_test=False).search(domain, limit=1)`  ?

Reply:
Same remark as in #1161

Reply:
When you think about it, you have to take into account the fact that you can have a space before and/or after the email in the 'email' field in the DB, because Odoo doesn't auto-strip char fields upon write(). That's why your proposal with a simple self.with_context(active_test=False).search(domain, limit=1) doesn't work. The '%' char in the domain matches any character and I'm not aware of any special char in the domain that would match only spaces (I would be very useful BTW).
Imagine you have 2 partners with emails :
1) '<space>alexis@via.fr'
2) 'alexis@via.fr.com'
With the code you propose, Odoo would show the duplicate warning. With my code, it won't show the duplicate warning.